sdk/test/trace/tracer_test.cc

```diff
@@ -12,6 +12,28 @@ using opentelemetry::core::SystemTimestamp;
 namespace nostd  = opentelemetry::nostd;
 namespace common = opentelemetry::common;
 
+/**
+ * A mock sampler that returns non-empty sampling results attributes.
+ */
+class MockSampler final : public Sampler
```

Comment:
Please move all sampler tests to a file `sampler_test.cc`.

Reply:
Do I need to push every test into one file? (including `always_on_test.cc`, etc?)

Reply:
I moved all sampler related test to one file.